### PR TITLE
add v4 fragment support

### PR DESCRIPTION
**Pull request by yaming116**

### CODE REVIEW

File: dart/pom.xml [hunk 1]
```diff
@@ -51,6 +51,11 @@
       <artifactId>android</artifactId>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>com.google.android</groupId>
+      <artifactId>support-v4</artifactId>
+      <version>${android-support.version}</version>
```
Comment (dlemures) [inline dart/pom.xml]: the dependency needs to be optional

**Comment by dlemures** [comment]
It failed, could you fix it?

**Comment by yaming116** [comment]
@dlemures  dependency to be optional build failed,i need remove it?

**Comment by dlemures** [comment]
It seems that none of the previous solutions work:
* compile dependency -> We don't want to include support library in every project
* optional / provided dependency -> It doesn't work in apps where the support library is not included, because Dart uses android.support.v4.app.Fragment

You will need to create a new sibling module (`dart-support`?) so if the app uses support fragment it will include dart-support instead of dart:
`compile 'com.f2prateek.dart:dart-support:XXX'`

One possible approach:
* Create class `com.f2prateek.dart.support.Dart` inside the new module dart-support
* `com.f2prateek.dart.support.Dart` extends from `com.f2prateek.dart.Dart` and adds the methods for support fragments
* for the enum, you can create a interface with the method `public Object getExtra(Object source, String key)` and both Dart classes define an enum which extends it.

What do you think?
@yaming116 @stephanenicolas

**Comment by dlemures** [comment]
I think it is a bit overkilling to make so many changes for a shortcut. From the support fragment we can always use:

`Dart.inject(this, getArguments)`

**Pull request closed without merging**